### PR TITLE
chore(releasing): Prep 0.14.0 release

### PR DESCRIPTION
**Pull request by jszwedko**
A few notes:

- I opted not to highlight emitting multiple events from `remap` yet as I'd really like to get the `unnest` PR in there first #7404 since it's not super useful until then.
- I opted not to highlight the new `redact` function until we add additional filters (#7435) and possibly redactors (#7445) since it is a bit lack luster until then.

TODOs:
- [x] Decide whether we want to / document the new `http` provider support from #7166. I pinged @leebenson about this.

Signed-off-by: Jesse Szwedko <jesse@szwedko.me>

### CODE REVIEW

File: docs/highlights/2021-06-02-fix-vector-metric-naming.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+last_modified_on: "2021-06-02"
+$schema: ".schema.json"
+title: "Telemetry units for duration metrics fixed"
+description: "A few telemetry metric names incorrectly had `nanoseconds_total` in their name when they are actually `seconds`"
+author_github: "https://github.com/jszwedko"
+pr_numbers: [7373]
+release: "0.14.0"
+hide_on_release_notes: false
+tags: ["type: announcement", "domain: metrics"]
```
Comment (binarylogic) [inline docs/highlights/2021-06-02-fix-vector-metric-naming.md:10]:
```suggestion
tags: ["type: breaking change", "domain: metrics"]
```

File: docs/reference/releases/0.14.0.cue [hunk 2]
```diff
@@ -0,0 +1,252 @@
+package metadata
+
+releases: "0.14.0": {
+	date:     "2021-06-03"
+	codename: ""
+
+	description: """
+		The Vector team is pleased to announce version 0.14.0!
+
+		This release a new `exec` source that can be used to execute arbitrary programs and use their output to generate
```
Comment (binarylogic) [inline docs/reference/releases/0.14.0.cue:10]:
```suggestion
		This release includes a new `exec` source that can be used to execute arbitrary programs and use their output to generate
```

**Review by binarylogic — APPROVED** [review]
Only 2 small issues. Looks good otherwise.

File: docs/highlights/2021-06-02-fix-vector-metric-naming.md (hunk 1 above)
Comment (jszwedko) [reply docs/highlights/2021-06-02-fix-vector-metric-naming.md:10]: Aha, thanks, I was wondering how that labeling happened.

**Review by spencergilbert — APPROVED** [review]

**Comment by jszwedko** [comment]
I talked with Lee and we are going to just soft-release the config HTTP provider support for now and highlight it later with a blog post and documentation.

**Pull request merged**